subsys/net/lib/sockets/sockets_select.c

```diff
@@ -0,0 +1,201 @@
+/*
+ * Copyright (c) 2018 Linaro Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <sys/time.h>
+
+#include <net/socket.h>
+
+/* Get size, in elements, of an array within a struct. */
+#define STRUCT_MEMBER_ARRAY_SIZE(type, field) ARRAY_SIZE(((type *)0)->field)
+
+/* Outputs results in predefined word_idx and bit_mask vars. */
+#define FD_SET_CALC_OFFSETS(set) { \
+	unsigned int b_idx = fd % (sizeof(set->bitset[0]) * 8); \
+	word_idx = fd / (sizeof(set->bitset[0]) * 8); \
+	bit_mask = 1 << b_idx; \
+	}
+
+void ZSOCK_FD_ZERO(zsock_fd_set *set)
+{
+	int i;
+
+	for (i = 0; i < ARRAY_SIZE(set->bitset); i++) {
+		set->bitset[i] = 0;
+	}
+}
+
+int ZSOCK_FD_ISSET(int fd, zsock_fd_set *set)
+{
+	u32_t word_idx, bit_mask;
+
+	if (fd < 0 || fd >= ZSOCK_FD_SETSIZE) {
+		return 0;
+	}
+
+	FD_SET_CALC_OFFSETS(set);
+
+	return (set->bitset[word_idx] & bit_mask) != 0;
+}
+
+void ZSOCK_FD_CLR(int fd, zsock_fd_set *set)
+{
+	u32_t word_idx, bit_mask;
+
+	if (fd < 0 || fd >= ZSOCK_FD_SETSIZE) {
+		return;
+	}
+
+	FD_SET_CALC_OFFSETS(set);
+
+	set->bitset[word_idx] &= ~bit_mask;
+}
+
+void ZSOCK_FD_SET(int fd, zsock_fd_set *set)
+{
+	u32_t word_idx, bit_mask;
+
+	if (fd < 0 || fd >= ZSOCK_FD_SETSIZE) {
+		return;
+	}
+
+	FD_SET_CALC_OFFSETS(set);
+
+	set->bitset[word_idx] |= bit_mask;
+}
+
+int zsock_select(int nfds, zsock_fd_set *readfds, zsock_fd_set *writefds,
+		 zsock_fd_set *exceptfds, struct timeval *timeout)
+{
+	struct pollfd pfds[CONFIG_NET_SOCKETS_POLL_MAX];
+	int i, res, poll_timeout;
+	int num_pfds = 0;
+	int num_selects = 0;
+	int fd_no = 0;
+
+	for (i = 0; i < STRUCT_MEMBER_ARRAY_SIZE(zsock_fd_set, bitset); i++) {
+		u32_t bit_mask = 1;
+		u32_t read_mask = 0, write_mask = 0, except_mask = 0;
+		u32_t ored_mask;
+
+		if (readfds != NULL) {
+			read_mask = readfds->bitset[i];
+		}
+
+		if (writefds != NULL) {
+			write_mask = writefds->bitset[i];
+		}
+
+		if (exceptfds != NULL) {
+			except_mask = exceptfds->bitset[i];
+		}
+
+		ored_mask = read_mask | write_mask | except_mask;
+
```

Comment:
usually there is no empty line between variable assigment and when checking its value, so this empty line can be removed.

Reply:
Fixed